src/usb_cam.cpp

```diff
@@ -311,7 +328,7 @@ bool UsbCam::read_frame()
       buf.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
       buf.memory = V4L2_MEMORY_MMAP;
 
-      if (-1 == xioctl(fd_, VIDIOC_DQBUF, &buf)) {
+      if (-1 == xioctl(fd_, static_cast<int>(VIDIOC_DQBUF), &buf)) {
```

Comment:
Can you please explain why you selected changing the prototype here?

Reply:
@twdragon this one actually confused me as well. It was again a compiler warning saying that it was warning about an overflow conversion.  You can see [the warnings here](https://github.com/ros-drivers/usb_cam/actions/runs/3502098184/jobs/5866269947#step:5:1378):

```
/__w/usb_cam/usb_cam/ros_ws/src/o43fy35fs5/usb_cam/src/usb_cam.cpp:286:29: warning: overflow in conversion from ‘long unsigned int’ to ‘int’ changes value from ‘3227014673’ to ‘-1067952623’ [-Woverflow]
    286 |       if (-1 == xioctl(fd_, VIDIOC_DQBUF, &buf)) {
```

Casting the inputs to the `xioctl` call resolved the issue, but honestly I dont really understand why this was a problem in the first place since the `xioctl` problem arg has the same `unsigned long` type as the input (at least I think)....

Reply:
@flynneva I found the following: 
```
/usr/include/linux/videodev2.h:2491:#define VIDIOC_DQBUF		_IOWR('V', 17, struct v4l2_buffer)
```
It is a macro declaring buffer exchange structure. I cannot see that there is a problem with pointer size, so I can only suppose that the compiler here does not deduct properly the type of literal. Did you try to cast to `unsigned long`? I think here this should be done, because `sizeof(int)` is not enough to address a multimedia buffer, as [kernel documentation](https://www.kernel.org/doc/html/latest/driver-api/ioctl.html) says.

Reply:
@flynneva can you test this fragment with `unsigned long` casting? In case of successful compiling, this PR for ROS2 should be approved

Reply:
@twdragon yes I just tried to cast to `unsigned long` locally instead and am still seeing the compiler warning. From [other xioctl examples online](https://jayrambhia.com/blog/capture-v4l2) it looks like everyone else uses `int` as well there...so I think it is OK.